File: api/src/endpoint/organization/list_organization_repositories.rs
use serde::{Deserialize, Serialize};

use api_derive::ApiRequest;

use crate::{endpoint::Endpoint, resource::RepositoryResource};

pub struct ListOrganizationRepositories;

impl Endpoint for ListOrganizationRepositories {
    const PATH: &'static str = "/organization/{org_name}/repositories";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListOrganizationRepositoriesRequest;
    type Response = ListOrganizationRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct ListOrganizationRepositoriesRequest {}

pub type ListOrganizationRepositoriesResponse = Vec<RepositoryResource>;

File: gitdot-api-ts/src/endpoint/organization/list-organization-repositories.ts
import { z } from "zod";

import { RepositoryResource } from "../../resource";

export const ListOrganizationRepositoriesRequest = z.object({});
export type ListOrganizationRepositoriesRequest = z.infer<
  typeof ListOrganizationRepositoriesRequest
>;

export const ListOrganizationRepositories = {
  path: "/organization/{org_name}/repositories",
  method: "GET",
  request: ListOrganizationRepositoriesRequest,
  response: z.array(RepositoryResource),
} as const;
export type ListOrganizationRepositories = typeof ListOrganizationRepositories;
